connection.go

```diff
@@ -29,6 +32,45 @@ type connection struct {
 	readTimeout time.Duration
 }
 
+func newTLSConnection(address string, ca, cert, key []byte, timeout, readTimeout time.Duration) (*connection, error) {
+	roots := x509.NewCertPool()
+	ok := roots.AppendCertsFromPEM(ca)
+	if !ok {
+		panic("failed to parse root certificate")
```

Comment:
why panic if we return error from the function? also below there is `log.panic`, here `panic`, should we unify?